homeassistant/components/xiaomi_miio/binary_sensor.py

```diff
@@ -0,0 +1,89 @@
+"""Support for Xiaomi Miio binary sensors."""
+from enum import Enum
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+
+from .const import (
+    CONF_DEVICE,
+    CONF_FLOW_TYPE,
+    CONF_MODEL,
+    DOMAIN,
+    KEY_COORDINATOR,
+    KEY_DEVICE,
+    MODELS_HUMIDIFIER_MJJSQ,
+)
+from .device import XiaomiCoordinatedMiioEntity
+
+ATTR_NO_WATER = "no_water"
+ATTR_WATER_TANK_DETACHED = "water_tank_detached"
+
+BINARY_SENSOR_TYPES = (
+    BinarySensorEntityDescription(
+        key=ATTR_NO_WATER,
+        name="Water Tank Empty",
+        icon="mdi:water-off-outline",
+    ),
+    BinarySensorEntityDescription(
+        key="water_tank_detached",
+        name="Water Tank Detached",
+        icon="mdi:flask-empty-off-outline",
+    ),
+)
+
+HUMIDIFIER_MJJSQ_BINARY_SENSORS = (ATTR_NO_WATER, ATTR_WATER_TANK_DETACHED)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Xiaomi sensor from a config entry."""
+    entities = []
+
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        model = config_entry.data[CONF_MODEL]
+        sensors = []
+        if model in MODELS_HUMIDIFIER_MJJSQ:
+            device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+            coordinator = hass.data[DOMAIN][config_entry.entry_id][KEY_COORDINATOR]
```

Comment:
`device` and `coordinator` will probably always be the same regardless of model.